core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -667,6 +677,11 @@ private void buildChannel() {
             throw CLUSTER.errorCreatingChannelFromConfigFile(DEFAULT_JGROUPS_CONFIGURATION_FILE, e);
          }
       }
+
+      if (props.containsKey(SOCKET_FACTORY)) {
```

Comment:
I believe this is overwriting the `SocketFactory` set in line 620.

Reply:
Fixed. I really would like to avoid the silly `Transport` abstraction we have and use a proper configuration object to pass things to the `JGroupsTransport`